spec/Conversions.tex

```diff
@@ -35,11 +35,18 @@ \section{Implicit Conversions}
       to the type of the corresponding formal argument, if the formal's
       intent is \chpl{in} or \chpl{const in} or an abstract intent
       (\rsec{Abstract Intents}) with the semantics of
-      \chpl{in} or \chpl{const in}.
-
-\item The formal argument of a function call is converted
-      to the type of the corresponding actual argument, if the formal's
-      intent is \chpl{out}.
+      \chpl{in} or \chpl{const in}. This conversion does not occur
+      for \chpl{type} arguments~(\rsec{Formal_Type_Arguments} even
+      if they specify a concrete type. It does not occur for generic
+      arguments~(\rsec{Generic_Functions}). It does
+      occur for param arguments~(\rsec{Formal_Parameter_Arguments})
+      with a specified type.
+
+% MPF: I don't see how this rule can be type correct
+%      and it also doesn't seem to be implemented right now
+%\item The formal argument of a function call is converted
```

Comment:
@vasslitvinov, Can you check if you think this should be removed?

Reply:
Right. This does not make sense to me. Let us remove this altogether.

What our implementation does is a separate issue, though, which we can discuss another time.

Reply:
Let me take this back. It IS meaningful to convert the formal to the actual. The catch is that this conversion happens upon return from the function. So I propose to leave it in, clarifying "upon return".

We are not planning to implement this rule, probably ever. However we can still put it in the spec.

Reply:
Why put it in the spec if we do not implement it?
The spec is not a transcription of what we have or will implement, it is more of an ideal language design.
Still, why bother? I suggest to allow this conversion on `out` because it makes the language more general i.e. permits more programs.

Reply:
Thanks for the discussion, for now I'll leave a comment in the spec noting it is not implemented, and I'll leave the rule in there for now. If we ever get around to revising `out` intents the way I revised `in` intents in this past release cycle, we could get this behavior pretty easily I think.